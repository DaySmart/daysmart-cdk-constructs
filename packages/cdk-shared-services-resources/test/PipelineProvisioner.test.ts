// import * as cdk from '@aws-cdk/core';
// import '@aws-cdk/assert/jest'
// import { PipelineProvisioner } from '../lib/PipelineProvisioner';
// import { arrayWith, objectLike, SynthUtils } from '@aws-cdk/assert';

// test('pipeline provisioner template', () => {
//     const stack = new cdk.Stack();
//     new PipelineProvisioner(stack, 'PipelineProvisioner', {
//         owner: 'you',
//         repo: 'my-code'
//     });
    
//     expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
// });