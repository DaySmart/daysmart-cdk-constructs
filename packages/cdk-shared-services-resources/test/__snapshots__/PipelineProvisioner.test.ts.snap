// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pipeline provisioner template 1`] = `
Object {
  "Resources": Object {
    "PipelineProvisionerPolicyD672C756": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "cloudformation:*",
                "codepipeline:*",
                "codebuild:*",
                "codedeploy:*",
                "codestar:*",
                "codestar-connections:*",
                "s3:List*",
                "s3:Get*",
                "s3:Describe",
                "iam:Attach*",
                "iam:Create*",
                "iam:Get*",
                "iam:List*",
                "iam:PassRole",
                "iam:Put*",
                "lambda:*",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "PipelineProvisionerPolicyD672C756",
        "Roles": Array [
          Object {
            "Ref": "PipelineProvisionerRoleB8C40D93",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "PipelineProvisionerProject812B4DD6": Object {
      "Properties": Object {
        "Artifacts": Object {
          "Type": "NO_ARTIFACTS",
        },
        "BadgeEnabled": false,
        "ConcurrentBuildLimit": 1,
        "EncryptionKey": "alias/aws/s3",
        "Environment": Object {
          "ComputeType": "BUILD_GENERAL1_SMALL",
          "Image": "aws/codebuild/standard:5.0",
          "ImagePullCredentialsType": "CODEBUILD",
          "PrivilegedMode": false,
          "Type": "LINUX_CONTAINER",
        },
        "Name": "my-code-pipeline-provisioner",
        "ServiceRole": Object {
          "Fn::GetAtt": Array [
            "PipelineProvisionerRoleB8C40D93",
            "Arn",
          ],
        },
        "Source": Object {
          "BuildSpec": "{
  \\"version\\": \\"0.2\\",
  \\"phases\\": {
    \\"install\\": {
      \\"runtime-versions\\": {
        \\"nodejs\\": 14
      },
      \\"commands\\": [
        \\"cd infrastructure/pipeline\\",
        \\"npm i\\"
      ]
    },
    \\"build\\": {
      \\"commands\\": [
        \\"npx cdk deploy --require-approval never\\"
      ]
    }
  }
}",
          "Location": "https://bitbucket.org/you/my-code.git",
          "ReportBuildStatus": true,
          "Type": "BITBUCKET",
        },
        "Triggers": Object {
          "FilterGroups": Array [
            Array [
              Object {
                "Pattern": "PUSH",
                "Type": "EVENT",
              },
            ],
          ],
          "Webhook": true,
        },
      },
      "Type": "AWS::CodeBuild::Project",
    },
    "PipelineProvisionerRoleB8C40D93": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "codebuild.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "Description": "Enables the pipeline provisioner CodeBuild projects to provision new CodePipelines",
        "RoleName": "pipeline-provisioner-role-my-code",
      },
      "Type": "AWS::IAM::Role",
    },
    "PipelineProvisionerRoleDefaultPolicy937EBC29": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "logs:CreateLogGroup",
                "logs:CreateLogStream",
                "logs:PutLogEvents",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":logs:",
                      Object {
                        "Ref": "AWS::Region",
                      },
                      ":",
                      Object {
                        "Ref": "AWS::AccountId",
                      },
                      ":log-group:/aws/codebuild/",
                      Object {
                        "Ref": "PipelineProvisionerProject812B4DD6",
                      },
                    ],
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":logs:",
                      Object {
                        "Ref": "AWS::Region",
                      },
                      ":",
                      Object {
                        "Ref": "AWS::AccountId",
                      },
                      ":log-group:/aws/codebuild/",
                      Object {
                        "Ref": "PipelineProvisionerProject812B4DD6",
                      },
                      ":*",
                    ],
                  ],
                },
              ],
            },
            Object {
              "Action": Array [
                "codebuild:CreateReportGroup",
                "codebuild:CreateReport",
                "codebuild:UpdateReport",
                "codebuild:BatchPutTestCases",
                "codebuild:BatchPutCodeCoverages",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    "arn:",
                    Object {
                      "Ref": "AWS::Partition",
                    },
                    ":codebuild:",
                    Object {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    Object {
                      "Ref": "AWS::AccountId",
                    },
                    ":report-group/",
                    Object {
                      "Ref": "PipelineProvisionerProject812B4DD6",
                    },
                    "-*",
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "PipelineProvisionerRoleDefaultPolicy937EBC29",
        "Roles": Array [
          Object {
            "Ref": "PipelineProvisionerRoleB8C40D93",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
  },
}
`;
